### PR TITLE
LegalDocuments: Use updated footer api to provide links & modals

### PR DESCRIPTION
**Pull request by lscharmer**
This PR changes the `LegalDocuments` to use the new footer API based on this PR: https://github.com/ILIAS-eLearning/ILIAS/pull/8359.

This PR fixes Mantis Bug: https://mantis.ilias.de/view.php?id=42339 (Show goto links in the footer for anonymous user)

### CODE REVIEW

File: components/ILIAS/UICore/classes/Screen/PageContentProvider.php [hunk 1]
```diff
@@ -173,7 +173,7 @@ private function buildTabTitle(): string
 
     public function getFooterModification(CalledContexts $screen_context_stack): ?FooterModification
     {
-        return $this->globalScreen()->layout()->factory()->footer()->withModification(fn(?Footer $footer): ?Footer => $this->dic['legalDocuments']->modifyFooter($footer));
+        return $this->globalScreen()->layout()->factory()->footer();
```
Comment (chfsx) [inline components/ILIAS/UICore/classes/Screen/PageContentProvider.php]: return `null` or remove the whole method `getFooterModification` would be better here.

**Review by chfsx — CHANGES_REQUESTED** [review]
LGTM in general, only one small change needed in `PageContentProvider`, I'd recommend to remove the method completely since a `FooterModification` is no longer needed.

File: components/ILIAS/UICore/classes/Screen/PageContentProvider.php (hunk 1 above)
Comment (lscharmer) [reply components/ILIAS/UICore/classes/Screen/PageContentProvider.php]: You are right. I removed the method.

**Comment by chfsx** [comment]
Thanks a lot! @thibsy I hope the change in components/ILIAS/UICore/classes/Screen/PageContentProvider.php is fine for you as well since this file is in UICore.

**Review by chfsx — APPROVED** [review]

**Review by thibsy — APPROVED** [review]

**Pull request merged**